### PR TITLE
Add Apache Drill Dialect to SqlGlot

### PR DESCRIPTION
**Pull request by cgivre**
This PR adds a dialect for Apache Drill to SQLGlot. 

Remaining work:

- [ ] Add support for Drill `table()` function
- [ ] Add unit tests for `table()` function.

### CODE REVIEW

File: sqlglot/dialects/drill.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        "''T''": "T",
+    }
+
+    class Tokenizer(tokens.Tokenizer):
+        QUOTES = ["'"]
+        IDENTIFIERS = ["`"]
+        ESCAPES = ["\\"]
+        ENCODE = "utf-8"
+
+        KEYWORDS = {**tokens.Tokenizer.KEYWORDS, "VARBINARY": TokenType.BINARY}
```
Comment (tobymao) [inline sqlglot/dialects/drill.py:115]: why did you add varbinary? that's already a type

**Comment by tobymao** [comment]
this is a pretty good start, why don't we get this merged in and then fix the xlsx table afterwards?

**Comment by cgivre** [comment]
> this is a pretty good start, why don't we get this merged in and then fix the xlsx table afterwards?

@tobymao I'd defer to you on that.   Drill's SQL dialect follows ANSI 2003 pretty closely, so I don't anticipate there's much more do to.  I'm happy to proceed if that's what you think is best.  Going to sleep now, but tomorrow, I'll remove the unit test for the `table()` function.

**Comment by tobymao** [comment]
ok yea, let's just get this merged. it's nice and clean. the table function is going to be much harder

File: sqlglot/dialects/drill.py (hunk 1 above)
Comment (cgivre) [reply sqlglot/dialects/drill.py:115]: Removed.  Drill doesn't have a `BINARY` data type, just a `VARBINARY`, so that's what I was trying to do here, but I think that is covered elsewhere.

Comment (cgivre) [reply sqlglot/dialects/drill.py]: Gotcha.... and it's already there. 👍 

https://github.com/tobymao/sqlglot/blob/f1f4fa2dfd9e20277e8496623c339ef9ee34b687/sqlglot/dialects/drill.py#L129-L132
